content/en/blog/_posts/2021-XX-XX-writing-a-controller-for-pod-labels.md

```diff
@@ -0,0 +1,473 @@
+---
+layout: blog
+title: "Writing a Controller for Pod Labels"
+date: 2021-03-XX
+slug: writing-a-controller-for-pod-labels
+---
+
+**Authors**: Arthur Busser, Padok
+
+[Operators][what-is-an-operator] are proving to be an excellent solution to
+running stateful distributed applications in Kubernetes. Open Source tools like
+the [Operator SDK][operator-sdk] provide ways to build reliable and maintainable
+operators, and extending Kubernetes to implement custom scheduling needs is now
+easier than ever.
+
+Kubernetes operators run complex software inside your cluster. The open-source
+community has already built [many operators][operatorhub] for distributed
+applications like Prometheus, ElasticSearch, or ArgoCD. Even outside of
+open-source, operators can help to bring new functionality to your Kubernetes
+cluster.
+
+An Operator is a set of [custom resources][custom-resource-definitions] and a
+set of [controllers][controllers] that, when combined, run complex software
+inside your Kubernetes cluster. A controller watches for changes to specific
+resources in the Kubernetes API and reacts by creating, updating, or deleting
+resources.
+
+The Operator SDK is best suited for building fully fledged operators.
+Nonetheless, you can use it to write a single controller. This post will walk
+you through writing a Kubernetes controller in Go that will add a `pod-name`
+label to Pods that have a specific annotation.
+
+## Why do we need a controller for this?
+
+I recently worked on a project where we needed to create a Service that routed
+traffic to a specific Pod in a ReplicaSet. The problem is that a Service can
+only select Pods by label, and all pods in a ReplicaSet have the same labels.
+There are two ways to solve this problem:
+
+1. Create a Service without a selector and manage the Endpoints or
+   EndpointSlices for that Service directly. We would need to write a custom
+   controller to insert our Pod's IP address into those resources.
+2. Add a label to the Pod a unique value. We could then use this label in our
+   Service's selector. Again, we would need to write a custom controller to add
+   this label.
+
+A controller is a control-loop that tracks one or more Kubernetes resource
+types. The controller from option n°2 above only needs to track Pods, which
+makes it simpler to implement. This is the option we are going to walk through
```

Comment:
Here, you start using "we" to mean me, the author, and you, the reader. I think your intention is to be polite. However, it would be more natural and personal to use "I and you". For example, "This is the option I'm going to walk you through..."
Later, you flip back and forth between "we" and "you" a bit. I think you can comfortably speak directly to the reader, like they're  a good friend or colleague who has asked you to show them how to do something difficult.

Reply:
The style guide recommends [addressing the reader as you](https://kubernetes.io/docs/contribute/style/style-guide/#address-the-reader-as-you).

Reply:
For a blog it's OK to use “we” provided it's clear from context who “we” are (eg a SIG, multiple authors, explicitly the author and reader together).

Reply:
Sorry, @busser. This is my first review of a post for the Kubernetes blog. I misapplied the style guide for technical documentation to these blog posts. You can ignore my suggestions about converting "we and let's" phrases to you and I.